solutions/usaco-1159.mdx

```diff
@@ -140,3 +144,197 @@ public class ConnectingTwoBarns {
 </JavaSection>
 
 </LanguageSection>
+
+## Alternative Implementation - DFS + Binary Search
+
+As the editorial mentioned, we can minimize the cost function by using binary search. Since each conncected component array is in sorted order (if we add the nodes of each connected component in increasing order), we can binary search on the sorted connected component array to find the closest field $j$ to for each field $i$ using `std::lower_bound`.
```

Comment:
```suggestion
As the editorial mentioned, we can minimize the cost function by using binary search. Since each conncected component array is in sorted order (if we add the nodes of each connected component in increasing order), we can binary search on the sorted connected component array to find the closest field $j$ for each field $i$ using `std::lower_bound`.
```